app/Http/Controllers/AdminController.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\AddAdminRequest;
+use App\Http\Requests\UpdateAdminRequest;
+use App\Http\Resources\PermissionResource;
+use App\Models\Admin;
+use App\Services\AdminService;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+
+class AdminController extends Controller
+{
+    protected $adminService;
+
+    public function __construct(AdminService $adminService)
+    {
+        $this->adminService = $adminService;
+    }
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(AddAdminRequest $request)
+    {
+        $this->authorize('create', Admin::class);
+        $this->adminService->store($request);
+        return response()->json(null, Response::HTTP_CREATED);
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        //
```

Comment:
Please write the show api as it's required to view the admin details when we edit him